src/webgl/shaders/immediateLight.vert

```diff
@@ -0,0 +1,70 @@
+precision mediump float;
+
+// vertex attributes
+attribute vec3 aPosition;
+attribute vec3 aNormal;
+attribute vec2 aTexCoord;
+
+// material properties
+attribute vec3 aAmbientColor;
+attribute vec4 aMaterialColor;
```

Comment:
nope, there's no way to do this dynamically. there's really not that much duplicated code - just the attribute/unfirom definitions. most of the lighting code is in `lighting.glsl`.

Reply:
I think it could be done if both parameters weren't called materialColor, but I have to think about this a little more when I can give it my full attention. I guess that could be done in a future patch but I'd like the public API for the shader uniforms and attributes to be as stable as possible.